core/storage/keystore/keystore.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "keystore.hpp"
+
+#include <primitives/address/impl/address_verifier_impl.hpp>
+
+#include "common/visitor.hpp"
+
+using fc::primitives::Protocol;
+using fc::storage::keystore::KeyStore;
+using fc::storage::keystore::KeyStoreError;
+
+KeyStore::KeyStore(std::shared_ptr<BlsProvider> blsProvider,
+                   std::shared_ptr<Secp256k1Provider> secp256K1Provider,
+                   std::shared_ptr<AddressVerifier> addressVerifier)
+    : bls_provider_(std::move(blsProvider)),
+      secp256k1_provider_(std::move(secp256K1Provider)),
+      address_verifier_(std::move(addressVerifier)) {}
+
+fc::outcome::result<bool> KeyStore::CheckAddress(
```

Comment:
`boost::get` [throws](https://www.boost.org/doc/libs/1_55_0/doc/html/boost/get_idp140956568.html), use `visit_in_place` [(example)](https://github.com/filecoin-project/cpp-filecoin/blob/master/core/primitives/address.cpp#L29) or catch exception

Reply:
`visit_in_place` is unable in current implementation. Added todos.